### PR TITLE
Add fatal error message for public signup disabled

### PR DESCRIPTION
**Pull request by IniZio**
ref #3596

Screenshot 2024-01-22 at 16-09-39 Authgear

### CODE REVIEW

**Comment by tung2744** [comment]
Please also update resources/authgear/templates/en/web/authflowv2/signup.html

**Comment by IniZio** [comment]
Updated to remove the condition in signup page

File: resources/authgear/templates/en/web/authflowv2/__error.html [hunk 1]
```diff
@@ -5,6 +5,7 @@
   {{ end }}
   {{- if $display_error -}}
     {{ $info := .Error.info }}
+    {{ .Error }}???
```
Comment (louischan-oursky) [inline resources/authgear/templates/en/web/authflowv2/__error.html:8]: Typo?

**Comment by louischan-oursky** [comment]
Now we will see this error `ERRO[0066] failed to create screen                       app=accounts error="public signup is disabled"` and it will become Sentry log. It is not desirable.

**Comment by louischan-oursky** [comment]
This manual exclusion is not very suitable. But let's accept this for now. I have created https://github.com/authgear/authgear-server/issues/3684

File: pkg/auth/handler/webapp/authflow_controller.go [hunk 2]
```diff
@@ -152,7 +153,10 @@ func (c *AuthflowController) HandleStartOfFlow(
 	if err != nil {
 		if errors.Is(err, authflow.ErrFlowNotFound) {
 			screen, err := c.createScreen(r, s, flowReference, input)
-			if err != nil {
+			if errors.Is(err, declarative.ErrNoPublicSignup) {
+				c.renderError(w, r, err)
```
Comment (tung2744) [inline pkg/auth/handler/webapp/authflow_controller.go:157]: Should update makeErrorResult to explicitly mark ErrNoPublicSignup as non recoverable error?

**Pull request merged**